modules/apps/document-library/document-library-preview-pdf/src/main/java/com/liferay/document/library/preview/pdf/internal/portlet/action/EditPDFPreviewMVCActionCommand.java

```diff
@@ -87,15 +89,42 @@ private void _updatePDFPreview(
 		long maxNumberPages = ParamUtil.getLong(
 			actionRequest, "maxNumberPages");
 
+		long systemMaxNumberPages =
+			_pdfPreviewManagedServiceFactory.getSystemMaxNumberPages();
+
 		if (scope.equals(
 				ExtendedObjectClassDefinition.Scope.COMPANY.getValue())) {
 
+			if ((systemMaxNumberPages != 0) &&
+				(systemMaxNumberPages < maxNumberPages)) {
+
+				throw new PDFPreviewException(systemMaxNumberPages);
+			}
+
 			_pdfPreviewManagedServiceFactory.updateCompanyPDFPreview(
 				scopePK, maxNumberPages);
 		}
 		else if (scope.equals(
 					ExtendedObjectClassDefinition.Scope.GROUP.getValue())) {
 
+			if ((systemMaxNumberPages != 0) &&
+				(systemMaxNumberPages < maxNumberPages)) {
+
+				throw new PDFPreviewException(systemMaxNumberPages);
+			}
+
+			Group group = _groupService.getGroup(scopePK);
+
+			long companyMaxNumberPages =
+				_pdfPreviewManagedServiceFactory.getCompanyMaxNumberPages(
+					group.getCompanyId());
+
+			if ((companyMaxNumberPages != 0) &&
+				(companyMaxNumberPages < maxNumberPages)) {
+
+				throw new PDFPreviewException(companyMaxNumberPages);
+			}
+
 			_pdfPreviewManagedServiceFactory.updateGroupPDFPreview(
 				scopePK, maxNumberPages);
```

Comment:
Can you create a new `updatePDFPreview` method that takes the scopePK, scope and that does all the validations internally?
```
public void updatePDFPreview(ExtendedObjectClassDefinition.Scope scope, long scopePK, ...) {
     if (scope.equals(...)) {
     }
     ...
}
```